grails-app/controllers/org/broadinstitute/orsp/UserController.groovy

```diff
@@ -29,4 +35,52 @@ class UserController {
         render syncedUsers as JSON
     }
 
+    def rolesManagement() {
+        render(view: "/rolesManagement/index")
+    }
+
+    def getOrspUsers() {
+        JSON.registerObjectMarshaller(User) {
+            def output = [:]
+            output['id'] = it.id
+            output['userName'] = it.userName
+            output['displayName'] = it.displayName
+            output['emailAddress'] = it.emailAddress
+            output['createdDate'] = it.createdDate
+            output['updatedDate'] = it.updatedDate
+            output['lastLoginDate'] = it.lastLoginDate
+            output['roles'] = it.roles.stream().map{role ->
+                role.getRole()}.collect()
+            return output
+        }
+
+        PaginationParams pagination = new PaginationParams(
+                draw: params.getInt("draw") ?: 1,
+                start: params.getInt("start")?: 0,
+                length: params.getInt("length")?: 10,
+                orderColumn: params.getInt("orderColumn")? params.get("orderColumn"): 0,
+                sortDirection: params.get("sortDirection")? params.get("sortDirection").toString() : "asc",
+                searchValue: params.get("searchValue")? params.get("searchValue").toString() : null)
+        render(queryService.queryUserRoles(pagination) as JSON)
+    }
+
+    def editOrspUserRole() {
+        Integer userId = request.JSON["userId"] as Integer
+        ArrayList<String> rolesToAssign = request.JSON["roles"] as ArrayList<String>
+
+        try {
+            userService.editUserRoles(userId, rolesToAssign)
+            response.status = 200
+            render([message: 'Role Updated'] as JSON)
+        }
+        catch(IllegalArgumentException e) {
+            response.status = 400
+            render([error: e.message] as JSON)
+        }
+        catch(Exception e) {
+            log.error("Error while trying to modify roles to userId: ${userId}." + e.message)
+            response.status = 500
+            render([error: e.message] as JSON)
+        }
```

Comment:
Might be safer catching `Throwable` here.

Reply:
I went through the code to see if I was missing any other possible error, so I included SQLException to UserService method.
I sticked to the idea of catching `Exception` at the end of the code for any un-logged problem but this shouldn't occur in this method.

Reply:
RE: Sql exceptions ... the content of those exceptions should never be returned raw, i.e. `render([error: e.message] as JSON)`, because they can contain relatively sensitive information about the query, etc. What I generally do is log that info and return something generic to the user.